dasharo-compatibility/usb-hid-and-msc-support.robot

```diff
@@ -74,12 +76,10 @@ USB002.001 USB keyboard detected in FW
 
 USB002.002 USB keyboard in OS (Ubuntu)
     [Documentation]    Check whether the external USB keyboard is detected
-    ...    correctly by the Linux OS and all basic keys work
-    ...    according to their labels.
-    IF    not ${USB_KEYBOARD_DETECTION_SUPPORT}
-        SKIP    USB002.002 not supported
-    END
-    IF    not ${TESTS_IN_UBUNTU_SUPPORT}    SKIP    USB002.002 not supported
+    ...    correctly by the Linux OS.
+    Skip If    not ${USB_KEYBOARD_DETECTION_SUPPORT}    USB002.002 not supported
+    Skip If    "${DEVICE_USB_KEYBOARD}" == "${EMPTY}"    USB002.002 not supported
```

Comment:
So we have:
- USB_KEYBOARD_DETECTION_SUPPORT
- DEVICE_USB_KEYBOARD
- HAS_KEYBOARD

Isn't it too much for nearly the same meaning?

Reply:
I'm not sure if those global variables have documentation anywhere.
From what I can understand `*_SUPPORT` variable means that it's possible to test this on this platform as a whole not that this particular platform has keyboard attached.
This test checks that `${DEVICE_USB_KEYBOARD}` is in output of `lsusb` but if this variable is empty test succeeds everytime so doesn't really test anything.
HAS_KEYBOARD is combination of DEVICE_USB_KEYBOARD and PiKVM (I'm assuming if PiKVM is connected then it should always detect it as keyboard?)

This PR mostly fixes reported issues and one I encountered during tests so I didn't make bigger changes

Reply:
fair enough, thanks for explaining